cartoframes/data/registry/base_dataset.py

```diff
@@ -172,13 +175,19 @@ def _get_query_columns(self):
         return Column.from_sql_api_fields(table_info['fields'])
 
     def _get_geom_type(self, query=None):
-        return get_query_geom_type(self._context, query or self.get_query())
+        return utils.get_query_geom_type(self._context, query or self.get_query())
 
     def _get_schema(self):
         if self._credentials:
-            return self._credentials.get_default_schema()
+            is_org_user = self._check_org_user()
+            return 'public' if not is_org_user else self._credentials.username
```

Comment:
It would be nice having a unique point for checking is the user is a org user. I would move `utils.is_org_user` method to context or credentials. We are doing more or less the same here and sql client. Also, I think we could avoid creating props like `self._is_org_user`

Reply:
Now, this utility function is used in both `BaseDataset` and `SQLClient`. Since it uses the new context I think the natural place to move this is to the new context.

Regarding the variable `self._is_org_user`. This is used as a "singleton" to avoid making the "is_org" request more than once. If `self._is_org_user` is defined it returns the variable, if it is not defined it makes the request (but only once).